File: chrome/browser/ui/android/autofill/autofill_wallet_reminder_notice_bottom_sheet_bridge.cc
// Copyright 2026 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/ui/android/autofill/autofill_wallet_reminder_notice_bottom_sheet_bridge.h"

#include "base/android/jni_android.h"
#include "components/autofill/android/payments/legal_message_line_android.h"
#include "ui/android/window_android.h"

// Must come after all headers that declare env/types.
#include "chrome/browser/autofill/android/jni_headers/AutofillWalletReminderNoticeBottomSheetBridge_jni.h"

namespace autofill {

AutofillWalletReminderNoticeBottomSheetBridge::
    AutofillWalletReminderNoticeBottomSheetBridge(
        ui::WindowAndroid* window_android) {
  if (window_android && window_android->GetJavaObject()) {
    JNIEnv* env = base::android::AttachCurrentThread();
    java_object_ =
        Java_AutofillWalletReminderNoticeBottomSheetBridge_Constructor(
            env, window_android->GetJavaObject());
  }
}

AutofillWalletReminderNoticeBottomSheetBridge::
    ~AutofillWalletReminderNoticeBottomSheetBridge() {
  if (java_object_) {
    JNIEnv* env = base::android::AttachCurrentThread();
    Java_AutofillWalletReminderNoticeBottomSheetBridge_destroy(env,
                                                               java_object_);
  }
}

void AutofillWalletReminderNoticeBottomSheetBridge::RequestShowContent(
    LegalMessageLines legal_message_lines) {
  if (java_object_) {
    JNIEnv* env = base::android::AttachCurrentThread();
    Java_AutofillWalletReminderNoticeBottomSheetBridge_requestShowContent(
        env, java_object_,
        LegalMessageLineAndroid::ConvertToJavaLinkedList(legal_message_lines));
  }
}

}  // namespace autofill

File: chrome/browser/autofill/android/java/src/org/chromium/chrome/browser/autofill/wallet_reminder_notice/AutofillWalletReminderNoticeBottomSheetBridge.java
// Copyright 2026 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.autofill.wallet_reminder_notice;

import android.content.Context;

import org.jni_zero.CalledByNative;
import org.jni_zero.JNINamespace;
import org.jni_zero.JniType;

import org.chromium.build.annotations.NullMarked;
import org.chromium.build.annotations.Nullable;
import org.chromium.components.autofill.payments.LegalMessageLine;
import org.chromium.components.browser_ui.bottomsheet.BottomSheetController;
import org.chromium.components.browser_ui.bottomsheet.BottomSheetControllerProvider;
import org.chromium.ui.base.WindowAndroid;

import java.util.List;

/** JNI bridge entry point to show the Wallet Reminder Notice bottom sheet. */
@JNINamespace("autofill")
@NullMarked
public class AutofillWalletReminderNoticeBottomSheetBridge {
    private final WindowAndroid mWindowAndroid;
    private @Nullable AutofillWalletReminderNoticeBottomSheetCoordinator mCoordinator;

    @CalledByNative
    public AutofillWalletReminderNoticeBottomSheetBridge(WindowAndroid windowAndroid) {
        mWindowAndroid = windowAndroid;
    }

    @CalledByNative
    public void requestShowContent(
            @JniType("std::vector") List<LegalMessageLine> legalMessageLines) {
        Context context = mWindowAndroid.getContext().get();
        BottomSheetController bottomSheetController =
                BottomSheetControllerProvider.from(mWindowAndroid);
        if (context == null || bottomSheetController == null) {
            return;
        }

        if (mCoordinator != null) {
            mCoordinator.destroy();
        }

        mCoordinator =
                new AutofillWalletReminderNoticeBottomSheetCoordinator(
                        context, bottomSheetController, legalMessageLines);
        mCoordinator.requestShowContent();
    }

    @CalledByNative
    public void destroy() {
        if (mCoordinator != null) {
            mCoordinator.destroy();
            mCoordinator = null;
        }
    }

    @Nullable AutofillWalletReminderNoticeBottomSheetCoordinator getCoordinatorForTesting() {
        return mCoordinator;
    }
}
